torch/utils/_pytree.py

```diff
@@ -137,6 +139,8 @@ def _register_pytree_node(
             flattened pytree.
         unflatten_fn: A callable that takes a flattened version of the pytree,
             additional context, and returns an unflattedn pytree.
+        serialized_type_name: A keyword argument used to specify the fully qualified
+            name used when serializing the tree spec.
```

Comment:
We've added a lot of fields to register_pytree_node. I've noticed that users have attempted to provide all of the things we ask for here.

We should make it clear (by adding a line above "Args" in the docstring) that the only required fields are typ, flatten_fn, and unflatten_fn. The other fields are optional and only necessary if someone wants to json serialize their type